src/tests/Microsoft.PowerFx.Interpreter.Tests/PowerFxEvaluationTests.cs

```diff
@@ -56,6 +57,46 @@ private static (RecalcEngine engine, RecordValue parameters) OptionSetTestSetup(
             return (new RecalcEngine(config), parameters);
         }
 
+        private static (RecalcEngine engine, RecordValue parameters) OptionSetSortTestSetup(PowerFxConfig config)
+        {
+            var optionSet = new OptionSet("OptionSet", DisplayNameUtility.MakeUnique(new Dictionary<string, string>()
+            {
+                    { "option_1", "Option1" },
+                    { "option_2", "Option2" }
+            }));
+
+            config.AddOptionSet(optionSet);
+
+            optionSet.TryGetValue(new DName("option_1"), out var o1Val);
+            optionSet.TryGetValue(new DName("option_2"), out var o2Val);
+
+            var r1 = FormulaValue.NewRecordFromFields(new NamedValue("OptionSetField1", o1Val), new NamedValue("StrField1", FormulaValue.New("test1")));
+            var r2 = FormulaValue.NewRecordFromFields(new NamedValue("OptionSetField1", o2Val), new NamedValue("StrField1", FormulaValue.New("test2")));
+            var r3 = FormulaValue.NewRecordFromFields(new NamedValue("OptionSetField1", o1Val), new NamedValue("StrField1", FormulaValue.New("test3")));
+            var r4 = FormulaValue.NewRecordFromFields(new NamedValue("OptionSetField1", o2Val), new NamedValue("StrField1", FormulaValue.New("test4")));
+            
+            // Testing with missing/blank option set field is throwing an exception. Once that is resolved uncomment and fix the test case in Sort.txt
+            var r5 = FormulaValue.NewRecordFromFields(new NamedValue("StrField1", FormulaValue.New("test5")));
+
+            var rType = RecordType.Empty()
+                .Add(new NamedFormulaType("OptionSetField1", FormulaType.OptionSetValue, "DisplayNameField1"))
+                .Add(new NamedFormulaType("StrField1", FormulaType.String, "DisplayNameField2"));
+
+            var t1 = new RecordsOnlyTableValue(Core.IR.IRContext.NotInSource(rType.ToTable()), new List<RecordValue>() { r1, r2 });
```

Comment:
>var t1 = new RecordsOnlyTableValue(Core.IR.IRContext.NotInSource(rType.ToTable()), new List<RecordValue>() { r1, r2 }); [](http://example.com/codeflow?start=12&length=119)

nit:  we have a public helper to simplify this
var t1 = FormulaValue.NewTable(rType, r1, r2);

(IRContext is internal)